src/components/ArtifactCollections.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div class="artifact-collections">
+    <div class="artifact-collections__filters">
+      <ResultsCount :label="localization.info.artifact" :count="artifactsCount" class="artifact-collection-list__results" />
+      <SearchInput v-model="searchTerm" :placeholder="localization.info.artifactSearch" :label="localization.info.artifactSearch" class="artifact-collection-list__search" />
+      <ArtifactTypeSelect v-model:selected="selectedType" class="artifact-collection-list__type" />
+      <ViewModeButtonGroup class="artifact-collections__view-mode-button-group" />
+    </div>
+
+    <template v-if="artifactsCountSubscription.executed && artifactsCount == 0">
+      <ArtifactCollectionsEmptyState />
+    </template>
+    <template v-else-if="artifacts && artifacts.length">
+      <RowGridLayout>
+        <template v-for="artifact in artifacts" :key="artifact.id">
+          <ArtifactCard :artifact="artifact" />
+        </template>
+      </RowGridLayout>
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { useSubscription } from '@prefecthq/vue-compositions'
+  import { computed, ref } from 'vue'
+  import { ArtifactTypeSelect, ResultsCount, SearchInput } from '@/components'
+  import ArtifactCard from '@/components/ArtifactCard.vue'
+  import ArtifactCollectionsEmptyState from '@/components/ArtifactCollectionsEmptyState.vue'
+  import RowGridLayout from '@/components/RowGridLayout.vue'
+  import ViewModeButtonGroup from '@/components/ViewModeButtonGroup.vue'
+  import { useWorkspaceApi } from '@/compositions'
+  import { localization } from '@/localization'
+  import { ArtifactsFilter, ArtifactType } from '@/models'
+
+  const searchTerm = ref('')
+  const selectedType = ref<ArtifactType | null>(null)
+
+  const api = useWorkspaceApi()
+
+  const artifactsFilter = computed<ArtifactsFilter>(() => {
+    return {
+      artifacts: {
+        keyLike: searchTerm.value ? `%${searchTerm.value}%` : undefined,
```

Comment:
Is the api expecting this percent formatted string? Any other `like` filters that I've used accept a plain string without and tokens.

Reply:
If that's the case, do you need to cast to undefined if its falsey? Or can you just send an empty string. OR better yet handle that in the mapper.